extensions/poetry/main.ts

```diff
@@ -146,74 +103,103 @@ Deno.exit(status.code);
 async function poetryCheckDryRun(
   subcommand: string,
   args: string[],
-): Promise<number> {
-  // Skip lockfile update on install, since it doesn't have the `--lock` flag.
-  if (subcommand !== "install") {
-    console.log(`[${green("phylum")}] Updating lockfile…`);
-
-    const status = PhylumApi.runSandboxed({
-      cmd: "poetry",
-      args: [subcommand, "-n", "--lock", ...args.map((s) => s.toString())],
-      exceptions: {
-        run: [
-          "./",
-          "/bin",
-          "/usr/bin",
-          "~/.pyenv",
-          "~/.local/bin/poetry",
-          "~/Library/Application Support/pypoetry",
-          "~/.local/share/pypoetry",
-        ],
-        write: [
-          "./",
-          "~/.cache/pypoetry",
-          "~/Library/Caches/pypoetry",
-          "~/.pyenv",
-        ],
-        read: [
-          "./",
-          "~/.cache/pypoetry",
-          "~/Library/Caches/pypoetry",
-          "~/.pyenv",
-          "~/Library/Preferences/pypoetry",
-          "~/.config/pypoetry",
-          "/etc/passwd",
-        ],
-        net: true,
-      },
-    });
-
-    // Ensure dry-run update was successful.
-    if (!status.success) {
-      console.error(`[${red("phylum")}] Lockfile update failed.\n`);
-      await abort(status.code ?? 255);
+) {
+  const result = PhylumApi.runSandboxed({
+    cmd: "poetry",
+    args: [
+      subcommand,
+      "--no-interaction",
+      "--dry-run",
+      ...args.map((s) => s.toString()),
+    ],
+    exceptions: {
+      run: [
+        "./",
+        "/bin",
+        "/usr/bin",
+        "~/.pyenv",
+        "~/.local/bin/poetry",
+        "~/Library/Application Support/pypoetry",
+        "~/.local/share/pypoetry",
+      ],
```

Comment:
```suggestion
        "~/.local/share/pypoetry",
        "~/.local/pipx/venvs/poetry",
      ],
```
This additional entry is needed for `poetry` instances that are [installed with `pipx`](https://python-poetry.org/docs/#installing-with-pipx). The same entry is needed in the `PhylumExt.toml` manifest file. At least, this entry is known to work on a macOS system with `pipx`; it may be different for other environments.